### PR TITLE
Feature/1807 keyevents

### LINKED ISSUE #1807: Datepicker: Enable on key event handlers that passes keycode
Currently it is not possible to get access to the internal input field via ref in react which makes it impossible to hook into i.e. keyboard events to do auto formatting for the input value.

### PR DESCRIPTION
**Pull request by danyball**
Fixes #1807 

# Done is when (DoD):
- [x] Modifications within components are reflected by tests.
- [x] A self-review of the code has been performed.
- [x] The modified component properties have been added to the typescript typings.
- [x] Potential design changes have been reviewed by a designer.
- [x] The modifications are well documented (README.md, etc.) and showcased in the stories.
- [x] The modifications are documented in the code, particularly in hard-to-understand areas (focus on **WHY**).
- [x] The modifications are shortly added to CHANGELOG.md with the issue number.
- [x] The modifications still work in IE.
- [x] The modifications are according to our [Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)

### CODE REVIEW

File: src/components/20-molecules/datepicker/index.js [hunk 1]
```diff
@@ -253,6 +254,7 @@ class AXADatepicker extends NoShadowDOM {
     // internal model state
     this.state = {};
     applyDefaults(this);
+    this.handleInputKeyUp = this.handleInputKeyUp.bind(this);
```
Comment (01231) [inline src/components/20-molecules/datepicker/index.js]: Is there a reason for two different names?
I personally would name both handleInputKeyUp and onInputKeyUp or handleInputfieldKeyUp and onInputfieldKeyUp

**Review by 01231 — CHANGES_REQUESTED** [review]

File: src/components/20-molecules/datepicker/react/DemoDatepickerOnInputfieldKeyUp.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  return (
+    <div>
+      <span id="datepicker-react-testoutput" />
+      <br />
+      <AXADatepickerReact
+        inputfield
+        id="datepicker-react-inputfield-on-date-change"
+        onInputfieldKeyUp={e => {
+          document.querySelector('#datepicker-react-testoutput').innerHTML =
+            e.key;
```
Comment (01231) [inline src/components/20-molecules/datepicker/react/DemoDatepickerOnInputfieldKeyUp.jsx:16]: it's a little confusing just to see a letter above the input field without additional text. I would add some text like: `last typed letter: xy`

**Review by 01231 — CHANGES_REQUESTED** [review]

**Review by raphaellueckl — APPROVED** [review]
lgtm

**Review by 01231 — APPROVED** [review]
LGTM

**Pull request merged**